### DISCUSSION TITLE
Why is carriage return not a supported record delimiter?

### DISCUSSION THREAD

**Discussion by mdk25** [MarkPflug/Sylvan#157] [General]
I have discovered some client CSV files that use \r for the record delimiter. Why was support for this delimiter excluded in favor of only \n and \r\n?

Thanks.

**Comment by MarkPflug** [comment]
I'm not aware of any modern operating system that will write a \r for a newline. If someone goes out of their way to write one, then they can go out of their way to fix it before trying to use my library.

A workaround you can implement is to create a custom TextReader implementation that is a wrapper around another TextReader. Your wrapper will replace all \r with \n before returning back to the caller.

**Reply by mdk25** [reply]
I see, thanks. I certainly would not be surprised if this client is using some ancient technology to generate their files. For now I'll push back on them to consider modernizing, but will take note of that potential workaround if it becomes necessary. 👍

**Comment by mdk25** [comment]
I was pleasantly surprised to see that this is now supported: #198
